crates/keth-run/src/main.rs

```diff
@@ -0,0 +1,88 @@
+use clap::Parser;
+use pyo3::prelude::*;
+use std::path::PathBuf;
+const CANCUN_FORK: u64 = 19_426_588;
+
+#[derive(Parser, Debug)]
+#[command()]
+struct Args {
+    /// Block number to prove (must be after Cancun fork)
+    #[arg(
+        value_name = "BLOCK_NUMBER",
+        value_parser = clap::value_parser!(u64).range(CANCUN_FORK..),
+        help = "Block number (should be higher than Cancun fork (19426588))"
+    )]
+    block_number: u64,
+
+    /// Directory to save proof artifacts
+    #[arg(long, value_name = "DIR", default_value = "output")]
+    output_dir: PathBuf,
+
+    /// Directory containing ZKPI JSON files
+    #[arg(long, value_name = "DIR", default_value = "data/1/eels")]
+    data_dir: PathBuf,
+
+    /// Path to compiled Cairo program
+    #[arg(long, value_name = "FILE", default_value = "build/main_compiled.json")]
+    compiled_program: PathBuf,
+
+    /// Generate Stwo proof instead of prover inputs
+    #[arg(long)]
+    stwo_proof: bool,
+
+    /// Path to save the Stwo proof (required when --stwo-proof is used)
+    #[arg(long, value_name = "FILE", required_if_eq("stwo_proof", "true"))]
+    proof_path: Option<PathBuf>,
+
+    /// Verify the Stwo proof after generation (only used with --stwo-proof)
+    #[arg(long)]
+    verify: bool,
+}
+
+// Check if the PYTHONPATH, PYTHONHOME, and PYO3_PYTHON variables are set, panics if any one is
+// missing
+fn check_env() {
+    dotenvy::dotenv().expect("Failed to load .env file");
+
+    dotenvy::var("PYTHONPATH").expect("PYTHONPATH environment variable not set");
+    dotenvy::var("PYTHONHOME").expect("PYTHONHOME environment variable not set");
```

Comment:
this should be necessary but I found this issue https://github.com/PyO3/pyo3/issues/3717 where it's mentionned that py discovery at runtime is not so good.